plugins/kotlin-dataframe/testData/box/sum.kt

```diff
@@ -0,0 +1,74 @@
+import org.jetbrains.kotlinx.dataframe.*
+import org.jetbrains.kotlinx.dataframe.annotations.*
+import org.jetbrains.kotlinx.dataframe.api.*
+import org.jetbrains.kotlinx.dataframe.io.*
+
+fun box(): String {
+    // multiple columns
+    val personsDf = dataFrameOf(
+        "name",
+        "age",
+        "city",
+        "weight",
+        "height",
+        "yearsToRetirement",
+        "workExperienceYears",
+        "dependentsCount",
+        "annualIncome"
+    )(
+        "Alice", 15, "London", 99.5, "1.85", 50, 0.toShort(), 0.toByte(), 0L,
+        "Bob", 20, "Paris", 140.0, "1.35", 45, 2.toShort(), 0.toByte(), 12000L,
+        "Charlie", 100, "Dubai", 75.0, "1.95", 0, 70.toShort(), 0.toByte(), 0L,
+        "Rose", 1, "Moscow", 45.33, "0.79", 64, 0.toShort(), 2.toByte(), 0L,
+        "Dylan", 35, "London", 23.4, "1.83", 30, 15.toShort(), 1.toByte(), 90000L,
+        "Eve", 40, "Paris", 56.72, "1.85", 25, 18.toShort(), 3.toByte(), 125000L,
+        "Frank", 55, "Dubai", 78.9, "1.35", 10, 35.toShort(), 2.toByte(), 145000L,
+        "Grace", 29, "Moscow", 67.8, "1.65", 36, 5.toShort(), 1.toByte(), 70000L,
+        "Hank", 60, "Paris", 80.22, "1.75", 5, 40.toShort(), 4.toByte(), 200000L,
+        "Isla", 22, "London", 75.1, "1.85", 43, 1.toShort(), 0.toByte(), 30000L,
+    )
+
+    // scenario #0: all numerical columns
+    val res0 = personsDf.sum()
+
+    val sum01: Int? = res0.age
```

Comment:
do all accessors become nullable? hmm, that doesn't seem ideal. @koperagen, do we have some compile-time

Reply:
no, but i think it can't be helped

Reply:
alternatively, do we really return nullable int for an empty dataframe? maybe we should throw an exception instead?
but we should be careful about GroupBy.sum

Reply:
See https://github.com/Kotlin/dataframe/issues/961

Reply:
For min/max/median/percentile, the result is nullable however, but only null if the input is empty.

@koperagen, is there a compile-time check for when a DF is empty or not? Otherwise we could generate the result-type more tightly.

Reply:
There's no such check

Reply:
> alternatively, do we really return nullable int for an empty dataframe? maybe we should throw an exception instead? but we should be careful about how it will affect GroupBy.sum

No we don't, the result of sum is never null, the sum of an empty Int column is 0. Even if we have a column of type `Nothing?` we have a non-null return type, defaulting to 0.0.

Reply:
> There's no such check

Okay then for those we can follow what I wrote here https://github.com/Kotlin/dataframe/pull/1148#discussion_r2058186955